lib/diagnostic.js

```diff
@@ -6,29 +6,63 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
+const normalizeWords = function (input) {
   /* your response here */
+  const words = input.split(' ')
+  for (let i = 0; i < words.length; i++) {
+    words[i] = words[i].toUpperCase().replace(/[^\w]/g, '')
+  }
+
+  return words
 }
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
+const uniqueWords = function (input) {
   /* your response here */
+  const words = []
+  for (const word of input) {
+    console.log(word)
+    if (words.indexOf(word) !== -1) {
+      words.push(word)
+    }
+  }
+  return words
 }
 
 // 3. Write a function that takes a string argument and returns the count of
 // words in a string.
 // Provide the *option* to count unique words instead of total words, by passing
 // a second argument.
 
-const wordCount = function (/* */) {
+const wordCount = function (params) {
   /* your response here */
+  const input = params[0]
+  const unique = params[1] // boolean
+  const words = normalizeWords(input)
```

Comment:
I see what you're getting at here, although I think what you would actually want to do is define two separate parameters and then access those arguments inside the function. Something like `wordCount(string, countUniques)`